cmd/upgrade/roles/cmd.go

```diff
@@ -150,42 +141,95 @@ func run(cmd *cobra.Command, argv []string) error {
 		os.Exit(1)
 	}
 
-	err = checkPolicyAndClusterVersionCompatibility(policyVersion, clusterUpgradeVersion)
+	env, err := ocm.GetEnv()
 	if err != nil {
-		reporter.Errorf("%v", err)
+		reporter.Errorf("Failed to determine OCM environment: %v", err)
 		os.Exit(1)
 	}
 
-	env, err := ocm.GetEnv()
+	managedPolicies, err := awsClient.HasManagedPolicies(cluster.AWS().STS().RoleARN())
 	if err != nil {
-		reporter.Errorf("Failed to determine OCM environment: %v", err)
+		r.Reporter.Errorf("Failed to determine if cluster has managed policies: %v", err)
+		os.Exit(1)
+	}
+	// TODO: remove once AWS managed policies are in place
+	if managedPolicies && env == ocm.Production {
+		r.Reporter.Errorf("Managed policies are not supported in this environment")
 		os.Exit(1)
 	}
 
-	creator, err := awsClient.GetCreator()
+	unifiedPath, err := aws.GetPathFromAccountRole(cluster, aws.AccountRoles[aws.InstallerAccountRole].Name)
 	if err != nil {
-		reporter.Errorf("Failed to get IAM credentials: %s", err)
+		r.Reporter.Errorf("Expected a valid path for '%s': %v", cluster.AWS().STS().RoleARN(), err)
 		os.Exit(1)
 	}
 
-	// Determine if interactive mode is needed
-	if !interactive.Enabled() && !cmd.Flags().Changed("mode") && !skipInteractive {
-		interactive.Enable()
+	credRequests, err := ocmClient.GetCredRequests(cluster.Hypershift().Enabled())
+	if err != nil {
+		r.Reporter.Errorf("Error getting operator credential request from OCM %s", err)
+		os.Exit(1)
 	}
 
-	if interactive.Enabled() && !skipInteractive {
-		mode, err = interactive.GetOption(interactive.Input{
-			Question: "Roles upgrade mode",
-			Help:     cmd.Flags().Lookup("mode").Usage,
-			Default:  aws.ModeAuto,
-			Options:  aws.Modes,
-			Required: true,
-		})
+	if managedPolicies {
+		mode, err = handleModeFlag(cmd, mode, skipInteractive)
```

Comment:
This can be done before here and reuse value for both managed policies and default flow in my opinion

Reply:
Done